### ISSUE TITLE
SwiftGridViewDelegate: columnIndex returns erratic values

### ISSUE THREAD

**Issue by sedwo** [nlampi/SwiftGridView#47]
When providing a simple grid data set, the delegate:
`func dataGridView(_ dataGridView: SwiftGridView, widthOfColumnAtIndex columnIndex: Int) -> CGFloat {`
returns column indeces that are outside the expected column range in the data, causing a `Fatal error: Index out of range` error.

![image](https://user-images.githubusercontent.com/1434449/56065372-a17c1880-5d42-11e9-963f-1d339136c8fd.png)

**Comment by nlampi** [comment]
Hi @sedwo, can you provide more details around the dataSource used? I am having trouble recreating the issue stated here. 
Thanks!

**Comment by sedwo** [comment]
I'm just using your `PrettyExample` project to confirm this.

[SwiftGridView-master.zip](https://github.com/nlampi/SwiftGridView/files/3087554/SwiftGridView-master.zip)

**Comment by sedwo** [comment]
Hi @nlampi , just curious if you got a chance to check on this?

**Comment by nlampi** [comment]
Hi @sedwo, 
Sorry for the delay, I was able to look into and find the root cause for this issue. There were situations where the height was requested through size methods and the width was being ignored. This meant erratic values were being sent and widths that were being ignored were requested. These items were updated to explicitly request height only and now bypass this invalid width request (in v0.7.1). Let me know if you have any issues going forward. 
Thanks!
